io.github.voxelum.XMCL.yaml

```diff
@@ -0,0 +1,77 @@
+app-id: io.github.voxelum.XMCL
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+command: xmcl-wrapper.sh
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=pulseaudio
+  - --device=dri
+  - --filesystem=home
+  - --filesystem=xdg-run/pipewire-0
+  - --filesystem=xdg-run/discord-ipc-0:ro
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=com.discordapp.Discord
+  - --talk-name=com.feralinteractive.GameMode
+  - --system-talk-name=org.freedesktop.DBus
+  - --system-talk-name=net.hadess.SwitcherooControl
+  - --env=ELECTRON_ENABLE_STACK_DUMPING=false
+  - --env=ELECTRON_ENABLE_LOGGING=false
+modules:
+  - name: xrandr
+    sources:
+      - type: archive
+        url: https://xorg.freedesktop.org/archive/individual/app/xrandr-1.5.3.tar.xz
+        sha256: f8dd7566adb74147fab9964680b6bbadee87cf406a7fcff51718a5e6949b841c
+    cleanup:
+      - /share/man
+      - /bin/xkeystone
+  - name: mesa-utils
+    buildsystem: simple
+    build-commands:
+      - cp /usr/bin/glxinfo /app/bin/ || true
+    sources: []
+  - name: XMCL
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - mkdir -p /app/opt/xmcl
+      - mkdir -p /app/share/applications
+      - mkdir -p /app/share/icons/hicolor
+      - tar xf xmcl-0.50.5-x64.tar.xz -C /app/opt/xmcl --strip-components=1
+      - chmod +x /app/opt/xmcl/xmcl
+      - mkdir -p /app/share/icons/hicolor/512x512/apps
+      - cp io.github.voxelum.xmcl.png /app/share/icons/hicolor/512x512/apps/io.github.voxelum.XMCL.png
+      - chmod 644 /app/share/icons/hicolor/512x512/apps/io.github.voxelum.XMCL.png
+      - cp xmcl.desktop /app/share/applications/io.github.voxelum.XMCL.desktop
+      - chmod 644 /app/share/applications/io.github.voxelum.XMCL.desktop
+      - sed -i 's|Exec=xmcl|Exec=/app/bin/xmcl-wrapper.sh|g' /app/share/applications/io.github.voxelum.XMCL.desktop
+      - sed -i 's|Icon=xmcl|Icon=io.github.voxelum.XMCL|g' /app/share/applications/io.github.voxelum.XMCL.desktop
+      - printf '#!/bin/sh\nexport LD_LIBRARY_PATH=/app/opt/xmcl:$LD_LIBRARY_PATH\nexport
```

Comment:
DO NOT generate script files like that. There is a source type "script" for that purpose.